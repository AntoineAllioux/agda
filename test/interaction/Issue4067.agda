postulate
  A  : Set
  F  : Set → Set
  P  : F A → Set
  id : F A → F A

f : (x : F A) → P x → P x
f _ p = p

⟨_⟩ : F A → F A
⟨ x ⟩ = x

{-# NOINLINE ⟨_⟩ #-}

record R (F : Set → Set) : Set₁ where
  field
    g : F A → F A

open R ⦃ … ⦄

postulate
  instance
    r : R F

-- In all three cases below the ⟨_⟩ should be retained in the goal type.

works : (x : F A) → P x
works x = f ⟨ x ⟩ {!!}

also-works : (x : F A) → P (id x)
also-works x = f (id ⟨ x ⟩) {!!}

fails : (x : F A) → P (g x)
fails x = f (g ⟨ x ⟩) {!!}
