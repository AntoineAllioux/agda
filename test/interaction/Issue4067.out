(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : P ⟨ x ⟩ ?1 : P (id ⟨ x ⟩) ?2 : P (R.g r ⟨ x ⟩) " nil)
((last . 1) . (agda2-goals-action '(0 1 2)))
